allennlp/modules/augmented_lstm.py

```diff
@@ -0,0 +1,171 @@
+        applies a fixed dropout mask per sequence to the recurrent connection of the
+        LSTM.
+    use_highway: bool, optional (default = True)
+        Whether or not to use highway connections between layers. This effectively involves
+        reparameterising the normal output of an LSTM as:
+            gate = sigmoid(W_x1 * x_t + W_h * h_t)
+            output = gate * h_t  + (1 - gate) * (W_x2 * x_t)
+    Return
+    ------
+    output_accumulator : PackedSequence
+        The outputs of the LSTM for each timestep. A tensor of shape
+        (batch_size, max_timesteps, output_size) where for a given batch
+        element, all outputs past the sequence length for that batch are
+        zero tensors.
+    """
+    def __init__(self,
+                 input_size: int,
+                 output_size: int,
+                 direction: str = "forward",
+                 recurrent_dropout_probability: float = 0.0,
+                 use_highway: bool = True) -> None:
+        super(AugmentedLstm, self).__init__()
+        self.input_size = input_size
+        self.output_size = output_size
+        self.direction = direction
+        self.use_highway = use_highway
+
+        if use_highway:
+            self.input_linearity = torch.nn.Linear(input_size, 6 * output_size, bias=True)
+            self.state_linearity = torch.nn.Linear(output_size, 5 * output_size, bias=True)
+        else:
+            self.input_linearity = torch.nn.Linear(input_size, 4 * output_size, bias=True)
+            self.state_linearity = torch.nn.Linear(output_size, 4 * output_size, bias=True)
+
+        self.recurrent_dropout_probability = recurrent_dropout_probability
+
+    def forward(self, inputs: PackedSequence,  # pylint: disable=arguments-differ
+                initial_state: Optional[Tuple[torch.Tensor, torch.Tensor]] = None):
+        """
+        Parameters
+        ----------
+        inputs : PackedSequence, required.
+            A tensor of shape (batch_size, num_timesteps, input_size)
+            to apply the LSTM over.
+
+        initial_state : Tuple[torch.Tensor, torch.Tensor], optional, (default = None)
+            A tuple (state, memory) representing the initial hidden state and memory
+            of the LSTM. Each tensor has shape (batch_size, output_dimension).
+
+        Returns
+        -------
+        A PackedSequence containing a torch.FloatTensor of shape
+        (batch_size, num_timesteps, output_dimension) representing
+        the outputs of the LSTM per timestep.
+        """
+        if not isinstance(inputs, PackedSequence):
+            raise ConfigurationError('inputs must be PackedSequence but got %s' % (type(inputs)))
+
+        inputs, batch_lengths = pad_packed_sequence(inputs, batch_first=True)
+        batch_size = inputs.size()[0]
+        total_timesteps = inputs.size()[1]
+        output_accumulator = Variable(inputs.data.new().resize_(batch_size,
```

Comment:
Why do you need to go from `inputs.data.new()` if you're just going to resize and fill?